app/controllers/application_pets_controller.rb

```diff
@@ -6,9 +6,18 @@ def show
   end
 
   def update
-    pet = Pet.find(params[:id])
-    pet.update(adoptable: "pending")
-    redirect_to "/pets/#{pet.id}"
+    @pet = Pet.find(params[:id])
+    app_pet = ApplicationPet.where(pet_id: @pet.id).pluck(:application_id)
+    applicant = Application.find(app_pet).pop
+    if @pet.adoptable == "adoptable"
```

Comment:
We could refactor this a little with my pet.adoptable? method that's now in the Pet model.

Reply:
Yeah, that sounds like a good plan.